app/javascript/components/UtmLinksPage/UtmLinkForm.tsx

```diff
@@ -96,31 +112,35 @@ export const UtmLinkForm = () => {
     [],
   );
 
-  React.useEffect(
-    () => setErrorInfo(null),
-    [title, destination, permalink, utmSource, utmMedium, utmCampaign, utmTerm, utmContent],
-  );
+  React.useEffect(() => setErrorInfo(null), [form.data]);
```

Comment:
Can you use Inertia's form helper to manage form errors? https://inertiajs.com/docs/v2/the-basics/forms#form-errors